### PR TITLE
[CONTINT-3946] Make sure to retry sbom scans when sending the result back to the channel fails

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR fixes an edge case to make sure we retry sbom scans when sending the result back to the channel.

### Motivation
Cover every case

### Additional Notes
This scenario is very unlikely to happen. The scan result channels have large buffer and the processing is very fast. The scanner is slow and has a buffer of size 300, which is less than what the scan results buffer have.

### Possible Drawbacks / Trade-offs
None

### Describe how to test/QA your changes
It's very difficult to produce a scenario where the `sendResult` function produces an error. I suppose you could by enabling sbom collection and creating a scenario where the host scan result handler is not working properly with a custom build

### CODE REVIEW

File: pkg/sbom/scanner/scanner_test.go [hunk 1]
```diff
@@ -240,6 +240,68 @@ func TestRetryLogic_Host(t *testing.T) {
...
+
+	// Make sure we recover
+	res := <-resultCh
+	assert.Equal(t, expectedResult.Report, res.Report)
+
+	// Make sure we don't receive anything afterward
+	select {
+	case res := <-resultCh:
+		t.Errorf("unexpected result received %v", res)
+	case <-time.After(4 * time.Second):
```
Comment (davidor) [inline pkg/sbom/scanner/scanner_test.go:302]: The problem that I see with this test is that it's going to wait 4 seconds here plus 6 in the Sleep above. I don't have enough context about this code to propose an alternative, so I'll let you decide what to do about it.

Comment (AliDatadog) [reply pkg/sbom/scanner/scanner_test.go]: I reduced it to a few ms for every tests. I think it's better.

File: pkg/sbom/scanner/scanner_test.go [hunk 2]
```diff
@@ -105,8 +105,8 @@ func TestRetryLogic_Error(t *testing.T) {
 
 	// Set up the configuration as the default one is too slow
 	cfg := config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
-	cfg.Set("sbom.scan_queue.base_backoff", "1s", model.SourceAgentRuntime)
-	cfg.Set("sbom.scan_queue.max_backoff", "3s", model.SourceAgentRuntime)
+	cfg.Set("sbom.scan_queue.base_backoff", "200ms", model.SourceAgentRuntime)
+	cfg.Set("sbom.scan_queue.max_backoff", "600s", model.SourceAgentRuntime)
```
Comment (davidor) [inline pkg/sbom/scanner/scanner_test.go:109]: Should this be ms ?

Comment (AliDatadog) [reply pkg/sbom/scanner/scanner_test.go]: Yes, good catch 👍 In any case it should run for at most 400ms.

**Review by davidor — APPROVED** [review]

**Review by YoannGh — APPROVED** [review]

**Pull request merged**